c/driver/postgresql/statement.cc

```diff
@@ -679,6 +679,7 @@ int TupleReader::GetNext(struct ArrowArray* out) {
     return EIO;
   }
 
+  is_finished_ = true;
```

Comment:
This seems wrong, we want to mark the reader as finished whether there is an error or not